src/frontend/screens/Settings/MapSettings/MapImportBottomSheet.tsx

```diff
@@ -0,0 +1,322 @@
+          variation: "outlined",
+        },
+      ]}
+    >
+      <TouchableOpacity onPress={onPressImport} style={styles.importButton}>
+        <View style={styles.importTextAndIcon}>
+          <MaterialIcon name="file-upload" size={30} color={MEDIUM_GREY} />
+          <Text style={styles.text}> {t(m.importFromFile)}</Text>
+        </View>
+        <Text style={[styles.text, { textAlign: "center" }]}>
+          {"( .mbtiles )"}
+        </Text>
+      </TouchableOpacity>
+    </BottomSheetContent>
+  );
+};
+
+/**
+ * Bottom sheet content for when an imported file is processing, checking the
+ * format etc. is correct (shows until the import endpoint returns with the
+ * import id). User is unable to close this sheet.
+ */
+const ImportingContent = () => {
+  return <BottomSheetContent title="Processing file..." buttonConfigs={[]} />;
+};
+
+/**
+ * Bottom sheet content for when there is an error with the selected file
+ */
+const FileErrorContent = ({ onPressClose }: { onPressClose: () => void }) => {
+  const { formatMessage: t } = useIntl();
+  return (
+    <BottomSheetContent
+      title={t(m.importErrorTitle)}
+      icon={<ErrorIcon color={RED} size={90} style={styles.errorIcon} />}
+      description={t(m.fileErrorDescription)}
+      buttonConfigs={[
+        {
+          onPress: onPressClose,
+          text: t(m.close),
+          variation: "outlined",
+        },
+      ]}
+    />
+  );
+};
+
+type BottomSheetStates = "select" | "importing" | "error";
+
+export interface MapImportBottomSheetMethods {
+  /** Open the bottom sheet by expanding it to its maximum snap point */
+  open: () => void;
+}
+
+/** Custom backdrop that will close sheet when pressed */
+const BackdropPressable = (props: BottomSheetBackdropProps) => {
+  return (
+    <BottomSheetBackdrop {...props} opacity={0.3} pressBehavior={"close"} />
+  );
+};
+
+/** Custom backdrop that will not close sheet when pressed */
+const BackdropNonPressable = (props: BottomSheetBackdropProps) => {
+  return (
+    <BottomSheetBackdrop {...props} opacity={0.3} pressBehavior={"none"} />
+  );
+};
+
+const MapImportBottomSheet = React.forwardRef<MapImportBottomSheetMethods, {}>(
+  (_props, ref) => {
+    const [state, setState] = React.useState<BottomSheetStates>("select");
+    const bottomSheetRef = React.useRef<BottomSheetMethods>(null);
+    const { snapPoints, updateSheetHeight } = useSnapPointsCalculator();
+    const { add: addMapImport } = useMapImportsManager();
+
+    React.useImperativeHandle(ref, () => {
+      return {
+        open: () => bottomSheetRef.current?.expand(),
+      };
+    });
+
+    const closeSheet = () => {
+      // Set state on close so when it opens again this is what you see
+      setState("select");
+      bottomSheetRef.current?.close();
+    };
+
+    const handlePressImport = async () => {
+      setState("importing");
+      let results;
+      try {
+        results = await DocumentPicker.getDocumentAsync();
```

Comment:
maybe? don't know off the top of my head if it's common to see mbtiles files that aren't suffixed with `.mbtiles`. since it's just a sqlite file i thought that there may be some really small edge case where it doesn't have the suffix when exported from elswhere. maybe better to just define the type like you're suggesting though

Reply:
ah, `type` accepts a MIME type and unfortunately, there [doesn't seem to be a registered one for mbtiles](https://www.iana.org/assignments/media-types/media-types.xhtml). Using sqlite3's doesn't seem to work, so will just leave this as is. We have error handling code both server and client-side so at least running into file type errors are somewhat mitigated